hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/customfees/CustomFixedFeeAssessor.java

```diff
@@ -0,0 +1,141 @@
+                if (couldTriggerCustomFees(denom, result.getExemptDebits())) {
+                    final var collector = fee.feeCollectorAccountId();
+                    if (sender.equals(collector)) {
+                        continue;
+                    }
+                    // This is a top-level fixed fee, not a fallback royalty fee
+                    assessFixedFee(feeMeta, sender, fee, result);
+                }
+            }
+        }
+    }
+
+    /**
+     * Fixed fee can be either a hbar or hts fee. This method assesses the fixed fee and adds the assessed fee to the
+     * {@link AssessmentResult}.
+     * @param feeMeta the custom fee metadata for the token
+     * @param sender the sender of the transaction, which will be payer for custom fees
+     * @param fee the custom fee to be assessed
+     * @param result the assessment result which will be used to create next level of transaction body
+     */
+    public void assessFixedFee(
+            final CustomFeeMeta feeMeta, final AccountID sender, final CustomFee fee, final AssessmentResult result) {
+        if (isPayerExempt(feeMeta, fee, sender)) {
+            return;
+        }
+        final var fixedFeeSpec = fee.fixedFeeOrThrow();
+        if (!fixedFeeSpec.hasDenominatingTokenId()) {
+            assessHbarFees(sender, fee, result);
+        } else {
+            assessHtsFees(sender, feeMeta, fee, result);
+        }
+    }
+
+    /**
+     * Assesses if the fixed fee is a hbar fee and adds the assessed fee debit and credit
+     * to the {@link AssessmentResult}.
+     * @param sender the sender of the transaction, which will be payer for custom fees
+     * @param hbarFee the hbar fee to be assessed
+     * @param result the assessment result which will be used to create next level of transaction body
+     */
+    private void assessHbarFees(
+            @NonNull final AccountID sender, @NonNull final CustomFee hbarFee, @NonNull final AssessmentResult result) {
+        final var collector = hbarFee.feeCollectorAccountId();
+        final var fixedSpec = hbarFee.fixedFee();
+        final var amount = fixedSpec.amount();
+
+        adjustHbarFees(result, sender, hbarFee);
+        // add all assessed fees for transaction record
+        result.addAssessedCustomFee(AssessedCustomFee.newBuilder()
+                .effectivePayerAccountId(sender)
+                .amount(amount)
+                .feeCollectorAccountId(collector)
+                .build());
+    }
+
+    /**
+     * Assesses if the fixed fee is a hts fee and adds the assessed fee debit and credit
+     * to the {@link AssessmentResult}.
+     * @param sender the sender of the transaction, which will be payer for custom fees
+     * @param chargingTokenMeta the custom fee metadata for the token
+     * @param htsFee the hts fee to be assessed
+     * @param result the assessment result which will be used to create next level of transaction body
+     */
+    private void assessHtsFees(
```

Comment:
I would suggest not using `HTS` in method names, since it's not clear to newcomers what this is

Reply:
Done